source/posix/thread.c

```diff
@@ -179,3 +196,19 @@ void aws_thread_current_sleep(uint64_t nanos) {
 
     nanosleep(&tm, &output);
 }
+
+int aws_thread_current_at_exit(aws_thread_atexit_fn *callback, void *user_data) {
+    if (!tl_wrapper) {
+        return AWS_OP_ERR;
```

Comment:
shouldn't we raise AWS_ERROR_THREAD_NOT_JOINABLE or something?

Reply:
Sure!